src/project-manager.ts

```diff
@@ -319,49 +331,60 @@ export class ProjectManager implements Disposable {
 	 * @param filePaths files to process (both absolute and relative paths are accepted)
 	 * @param maxDepth stop collecting when reached given recursion level
 	 * @param seen tracks visited files to avoid cycles
+	 * @param childOf OpenTracing parent span for tracing
 	 */
-	private async ensureTransitiveFileDependencies(filePaths: string[], maxDepth: number, seen = new Set<string>()): Promise<void> {
-		filePaths = filePaths.filter(f => !seen.has(f));
-		if (filePaths.length === 0) {
-			return Promise.resolve();
-		}
-		filePaths.forEach(f => seen.add(f));
-
-		const absFilePaths = filePaths.map(f => util.toUnixPath(util.resolve(this.rootPath, f)));
-		await this.ensureFiles(absFilePaths);
-
-		if (maxDepth > 0) {
-			const importPaths = new Set<string>();
-			await Promise.all(filePaths.map(async filePath => {
-				const config = this.getConfiguration(filePath);
-				await config.ensureBasicFiles();
-				const contents = this.getFs().readFile(filePath);
-				const info = ts.preProcessFile(contents, true, true);
-				const compilerOpt = config.getHost().getCompilationSettings();
-				for (const imp of info.importedFiles) {
-					const resolved = ts.resolveModuleName(util.toUnixPath(imp.fileName), filePath, compilerOpt, config.moduleResolutionHost());
-					if (!resolved || !resolved.resolvedModule) {
-						// This means we didn't find a file defining
-						// the module. It could still exist as an
-						// ambient module, which is why we fetch
-						// global*.d.ts files.
-						continue;
+	private async ensureTransitiveFileDependencies(filePaths: string[], maxDepth: number, seen = new Set<string>(), childOf = new Span()): Promise<void> {
+		const span = childOf.tracer().startSpan('Ensure file imports', { childOf });
```

Comment:
not sure if it's good idea to trace these calls, there may be plenty. Won't we pollute logs?